### PR TITLE
feat: allow users to specify zero optim search space and runner config

### PR DESCRIPTION
**Pull request by garrett361**
## Description
New flags for greater user control (necessary for basic functionality)

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: ds_autotuning_prototype/core_api/ffn_example_ds_launcher/zero_search_config.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"1": {"reduce_bucket_size": [20000000.0, 2000000000.0, 20000000000.0], "allgather_bucket_size": [20000000.0, 2000000000.0, 20000000000.0]}, "2": {"allgather_bucket_size": [30000000.0, 3000000000.0, 30000000000.0]}}
```
Comment (tayritenour) [inline ds_autotuning_prototype/core_api/ffn_example_ds_launcher/zero_search_config.json:1]: Is this necessary or could we remove this? To the best of my understanding it's just a duplicate of the yaml by the same name

File: ds_autotuning_prototype/trial_classes/cifar10_moe_fake_data/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+## Data
+The CIFAR-10 dataset is downloaded from https://www.cs.toronto.edu/~kriz/cifar.html.
+
+## To Run
+If you have not yet installed Determined, installation instructions can be found
+under `docs/install-admin.html` or at https://docs.determined.ai/latest/index.html
+
+Run the following command: 
+```
+det experiment create moe.yaml .
```
Comment (tayritenour) [inline ds_autotuning_prototype/trial_classes/cifar10_moe_fake_data/README.md:23]: Just a nit, but we could also add something about how to call this with the `python3 -m determined.pytorch.deepspeed.dsat.autotune moe.yaml .` style

File: ds_autotuning_prototype/trial_classes/cifar10_moe_fake_data/ds_config.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    "enabled": true,
+    "fp16_master_weights_and_grads": false,
+    "loss_scale": 0,
+    "loss_scale_window": 500,
+    "hysteresis": 2,
+    "min_loss_scale": 1,
+    "initial_scale_power": 15
+  },
+  "wall_clock_breakdown": false,
+  "zero_optimization": {
```
Comment (tayritenour) [inline ds_autotuning_prototype/trial_classes/cifar10_moe_fake_data/ds_config.json:33]: Question: In this case, if we use the autotuning, is this optimization dictionary ignored?

**Review by tayritenour — APPROVED** [review]
LGTM. Just small questions and nits

File: ds_autotuning_prototype/core_api/ffn_example_ds_launcher/zero_search_config.json (hunk 1 above)
Comment (garrett361) [reply ds_autotuning_prototype/core_api/ffn_example_ds_launcher/zero_search_config.json:1]: it's a test that you can provide the same config with either a `json` or `yaml` file. All of the examples are just scratchwork at this point, basically.

File: harness/determined/pytorch/deepspeed/dsat/_utils.py [hunk 4]
```diff
@@ -1,23 +1,47 @@
...
 from determined.pytorch.deepspeed.dsat import _defaults
+from determined.util import merge_dicts
 
 
-def get_config_dict_from_yaml_path(path: str) -> Dict[str, Any]:
-    config = yaml.YAML(typ="safe")
-    with open(path, "r") as f:
-        config_dict: dict = config.load(f)
-    return config_dict
+# TODO: move this to determined.util?
```
[the remark this answers is not in the document]
Comment (garrett361) [reply harness/determined/pytorch/deepspeed/dsat/_utils.py:17]: I think we must, but couldn't find it. The function I wrote here is gross.

File: ds_autotuning_prototype/trial_classes/cifar10_moe_fake_data/README.md (hunk 2 above)
Comment (garrett361) [reply ds_autotuning_prototype/trial_classes/cifar10_moe_fake_data/README.md:23]: yeah, for sure will do that in all of the examples we edit.  This example is basically scratch work, though.

File: ds_autotuning_prototype/trial_classes/cifar10_moe_fake_data/ds_config.json (hunk 3 above)
Comment (garrett361) [reply ds_autotuning_prototype/trial_classes/cifar10_moe_fake_data/ds_config.json:33]: Depends, but potentially, yes.

With this PR you can just search over a subspace of the available `zero_optimization` keys. So, like you could only search over `reduce_bucket_size` and `dsat` will overwrite that field, but all of the other values in the present config would remain as they are here.

**Pull request merged**